overlord/devicestate/handlers_install.go

```diff
@@ -1284,6 +1286,174 @@ func (m *DeviceManager) loadAndMountSystemLabelSnaps(systemLabel string) (
 	return sys, snapInfos, snapSeeds, mntPtForType, unmount, nil
 }
 
+func structureFromName(name string, Structures []gadget.VolumeStructure) (*gadget.VolumeStructure, error) {
+	var foundStruct *gadget.VolumeStructure
+	for _, structure := range Structures {
+		// Future go versions will have per-iter variables... but not 1.13
+		structure := structure
+		if structure.Name == name {
+			if foundStruct != nil {
+				return nil, fmt.Errorf("duplicated structure with name %q", name)
+			}
+			foundStruct = &structure
+		}
+	}
+
+	if foundStruct == nil {
+		return nil, fmt.Errorf("structure with name %q not found", name)
+	}
+
+	return foundStruct, nil
+}
+
+func compareOffsets(offset1, offset2 *quantity.Offset) error {
+	if (offset1 == nil && offset2 != nil) || (offset1 != nil && offset2 == nil) {
+		return fmt.Errorf("cannot compare offsets, one is nil")
+	}
+	if offset1 != nil && offset2 != nil {
+		if *offset1 != *offset2 {
+			return fmt.Errorf("offset %d does not match %d from gadget.yaml", *offset1, *offset2)
```

Comment:
nitpick (also for `compareOffsetWrite` below): if we assume that `offset2` is the one from `gadget.yaml`, I'd rename it to `offsetGadget` or just `gadget`. Or maybe add an additional parameter which is a string that describes where the second offset comes from, and then you build the error message using this information?

I'm probably overthinking it, but it feels a bit strange to have a function with such a generic name that emits error messages specific to the caller's implementation.